cEP-0019.md

```diff
@@ -0,0 +1,208 @@
+Meta-review System
+===========================
+
+|Metadata|                                   |
+|--------|-----------------------------------|
+|cEP     |0019                               |
+|Version |1.0                                |
+|Title   |Meta-review System                 |
+|Authors |Boxuan Li <liboxuan@connect.hku.hk>|
+|Status  |Proposed                           |
+|Type    |Process                            |
+
+Abstract
+--------
+
+This cEP describes the details of the process of implementing the meta-review
+system as a part of the
+[GSoC'18 project](https://summerofcode.withgoogle.com/projects/#5188493739819008).
+
+Background
+----------
+
+People including the author of pull request respond to comments by attaching
+emojis. Those emojis are called meta-review, or
+[reactions](https://developer.github.com/v4/enum/reactioncontent/). THUMBS_UP
+and THUMBS_DOWN, together with other reactions, are used by people especially
+the author of pull request to publicly acknowledge or reject review comments.
+This provides feedback to reviewers and all other members.
+
+However, those reactions spread over all PRs and are not collected and
+analyzed. By tracking reactions, a nice feedback loop would be created
+for both senior and junior reviewers.
+
+Introduction
+------------
+
+The meta-review project is about building a meta-review system,
+similar to a meta-moderation system, that can track emoji responses to
+review comments. These responses are to be collected, processed and displayed
+on [gh-board](https://github.com/coala/gh-board), which is a nice serverless
+kanban board.
+
+The scope of this project would exclude non-public meta-reviews. That is,
+there would not be any anonymous meta-reviews as that introduces too many
+additional complexities. The public only version would be used for
+a long time before adding support for non-public meta-reviews.
+
+Objectives
+----------
+
+One goal of this project is to encourage people to do meta-reviews,
+systematically.
+
+Also, to encourage people to do more meta-reviews, statistics of meta-reviews
+are to be collected and analysed. People who do meta-reviews will get scores
+according to some metrics. A ranking list is to be displayed on gh-board.
+
+To conclude, this project builds a meta-review system on gh-board to display
+statistics (score ranking, meta-reviews in need), and encourage people to
+do meta-review.
+
+Proposed Approach
+-----------
+
+### Collection of Meta-reviews
+
+Meta-review info, namely reactions, can be fetched via GitHub API V4 GraphQL.
+A Sample query is as follows:
```

Comment:
Sample -> sample

Also include the returned result data for this sample query, as that will help the reader understand it.